packages/stack/src/views/Stack/StackView.tsx

```diff
@@ -328,7 +329,18 @@ export default class StackView extends React.Component<Props, State> {
     ) {
       // If route isn't present in current state, but was closing, assume that a close animation was cancelled
       // So we need to add this route back to the state
-      navigation.navigate(route);
+      navigation.dispatch((state) => {
+        const routes = [
+          ...state.routes.filter((r) => r.key !== route.key),
+          route,
+        ];
+
+        return CommonActions.reset({
+          ...state,
+          routes,
+          index: routes.length - 1,
+        });
+      });
```

Comment:
Previous code `navigation.navigate(route);` worked because the route has a `key` prop and because we're removing navigate by key in this PR you had to kinda reimplement this in place. Smart! But I would be much happier if there would be a unit test covering that if possible 😅

Reply:
I don't think it's currently possible to have a unit test for this, most of the logic in stack doesn't have tests right now and adding tests would require a lot of work.